catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -224,22 +238,32 @@ private Object configure() {
   @SuppressWarnings(
       "squid:S1172" /* The code parameter is required in blueprint-cm-1.0.7. See https://issues.apache.org/jira/browse/ARIES-1436. */)
   public void destroy(int code) {
-    if (routeCollection == null) {
-      return;
-    }
+    synchronized (initState) {
```

Comment:
This destroy method is called by blueprint with a separate thread, separate from the initialization thread. If initialization of a CDM configuration is still occurring, this will block until initialization is done, acquire the lock, and then destroy the CDM config based on the state of initialization in CdmState.